.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -143,90 +153,118 @@ jobs:
+      - name: Generate builder binary
+        id: generate
+        uses: slsa-framework/slsa-github-generator/.github/actions/generate-builder@main
+        with:
+          repository: "${{ needs.detect-env.outputs.repository }}"
+          ref: "${{ needs.detect-env.outputs.ref }}"
+          go-version: 1.19
+          # Note: This must be the non-randomized binary name, so that it can be downloaded from the release assets.
+          binary: "${{ env.BUILDER_BINARY }}"
+          compile-builder: "${{ inputs.compile-builder }}"
+          directory: "${{ env.BUILDER_DIR }}"
+          allow-private-repository: ${{ inputs.private-repository }}
+
+      - name: Upload builder
+        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # tag=v3.1.1
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          if-no-files-found: error
+          retention-days: 5
 
   ###################################################################
   #                                                                 #
   #     Invoke the builder binary to create a buildDefinition       #
   #                                                                 #
   ###################################################################
   generate-build-definition:
-    # TODO: Use the builder binary.
     outputs:
       build-definition-name: ${{ steps.generate.outputs.build-definition-name }}
       build-definition-sha256: ${{ steps.upload.outputs.sha256 }}
     runs-on: ubuntu-latest
-    needs: [detect-env, verify-builder-image]
+    needs: [rng, detect-env, verify-builder-image, generate-builder]
     steps:
+      - name: Checkout builder repository
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-builder-checkout@main
+        with:
+          repository: "${{ needs.detect-env.outputs.repository }}"
+          ref: "${{ needs.detect-env.outputs.ref }}"
+          path: __BUILDER_CHECKOUT_DIR__
+
+      - name: Download builder
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-download-artifact
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          sha256: "${{ needs.generate-builder.outputs.builder-binary-sha256 }}"
+          set-executable: true
+
+      - name: Get source ref
+        id: resolve
+        env:
+          SOURCE_DIGEST: ${{ inputs.source-digest }}
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          if [ -z "$SOURCE_DIGEST" ]; then 
+            ref=$(echo ${SOURCE_DIGEST} | cut -d':' -f2)
+            echo "ref=$ref" >> $GITHUB_OUTPUT
+          else 
+            echo "ref=$GITHUB_SHA" >> $GITHUB_OUTPUT
+          fi
+
+      # TODO: Use secure-checkout, but this requires configuring the source-repository.
```

Comment:
is this already done via `ref: ${{ steps.resolve.outputs.ref }}`?

Reply:
I think the issue is that this workflow takes a `source-repository` input which we don't do for our other workflows in order to make sure we check out the right project repository.

@asraa @rbehjati I'm not sure I understand the need for the `source-repository` input? Why isn't the `github.repository` always used?

Reply:
Yes I'm actually not sure that it's needed as a workflow level input. I understand it's useful to have the builder binary the source repository/source digest arguments, but inside the workflow, I don't expect this to differ.

I'll discuss online with @rbehjati today. If we remove, I'll update to clean that up.

Reply:
Yes. The source repo should be fetched from the GitHub context. But it is needed as an input to the builder command, because it has to be included in the `BuildDefinition`.

Reply:
Removed from workflow inputs, so we always use the default context.